consensus/src/threshold_simplex/actors/voter/actor.rs

```diff
@@ -180,102 +172,66 @@ impl<
     async fn add_verified_notarize(
         &mut self,
         public_key_index: u32,
-        notarize: Parsed<wire::Notarize, D>,
+        notarize: Notarize<D>,
     ) -> bool {
-        // Get proposal
-        let proposal = notarize.message.proposal.as_ref().unwrap();
-
-        // Compute proposal digest
-        let message = proposal_message(proposal.view, proposal.parent, &notarize.digest);
-        let proposal_digest = hash(&message);
-
         // Check if already notarized
-        if let Some(previous_notarize) = self.notaries.get(&public_key_index) {
-            if previous_notarize == &proposal_digest {
-                trace!(
-                    view = self.view,
-                    signer = public_key_index,
-                    ?previous_notarize,
-                    "already notarized"
-                );
+        if let Some(previous) = self.notarizes[public_key_index as usize].as_ref() {
+            if previous == &notarize {
+                trace!(?notarize, ?previous, "already notarized");
                 return false;
             }
 
             // Create fault
-            let previous_notarize = self
-                .notarizes
-                .get(previous_notarize)
-                .unwrap()
-                .get(&public_key_index)
-                .unwrap();
-            let previous_proposal = previous_notarize.message.proposal.as_ref().unwrap();
-            let proof = Prover::<D>::serialize_conflicting_notarize(
-                self.view,
-                previous_proposal.parent,
-                &previous_notarize.digest,
-                &previous_notarize.message.proposal_signature,
-                proposal.parent,
-                &notarize.digest,
-                &notarize.message.proposal_signature,
-            );
-            self.supervisor.report(CONFLICTING_NOTARIZE, proof).await;
+            let activity = ConflictingNotarize::new(previous.clone(), notarize);
+            self.reporter
+                .report(Activity::ConflictingNotarize(activity))
+                .await;
             warn!(
                 view = self.view,
                 signer = public_key_index,
-                activity = CONFLICTING_NOTARIZE,
                 "recorded fault"
             );
```

Comment:
Wonder if it makes sense to either:
- move the logging behavior into `.report()`
- change the message "recorded fault" to include the deleted information about the activity (CONFLICTING_NOTARIZE)

Reply:
Yeah, I think there is a cleanup to be had here.